src/workers/zig.ts

```diff
@@ -71,7 +71,8 @@ async function run(source: string) {
 
     const {libStd, wasm} = await wasmData;
 
-    let args = ["zig.wasm", "build-exe", "main.zig", "-Dtarget=wasm32-wasi"];
+    // The explicit -fno-llvm -fno-lld is a workaround for https://github.com/ziglang/zig/issues/16586
```

Comment:
Thanks, I updated the wording to be a bit more generic; I suspect similar issues will continue to occur until the work done on allowing LLVM to be more decoupled is further along (since that's what results in the internal compiler confusion when these flags are left out).